### PR TITLE
Compatibility with geo-types 0.4.0 to 0.6.0, adding unit tests for geo-types polygon conversion

### PR DESCRIPTION
**Pull request by rfilmyer**
I tested the library for compatibility from version 0.4.0 up through version 0.6.0 (the latest version). In order to do that, I also added some basic unit tests for polygon conversion, looking at converting `shapefile::Polygon`s to `geo_types::Polygon`s and back. I tested simple convex polygons and a polygon with a single inner ring. Additionally, geo-types `0.6.0` added 2 new `Geometry` structs: `Rect<T>` and `Triangle<T>`. I added a catchall clause to a function to error out on these types; we can implement them, but I didn't want to break compat for 0.4-0.5.

Future work could be adding on the various polygon classes (`PolygonM` and `PolygonZ`), as well as testing more complex polygons (with multiple holes, or with exclaves, or both).

Note that I uncovered a glitch in how the library converts polygons with inner rings from `shapefile::Polygon` to `geo_types::MultiPolygon<f64>`: the order appears to be reversed. Thus I had to write that test as follows:

```rust
    #[test]
    fn shapefile_polygon_to_geotypes_polygon_with_inner_ring() {

        let one_ring_polygon = Polygon::with_rings(vec![
            PolygonRing::Outer(vec![
                Point::new(-1.1,-1.01),
                Point::new(-1.2, 1.02),
                Point::new( 1.3, 1.03),
                Point::new( 1.4,-1.04),
                Point::new(-1.1,-1.01),
            ]), PolygonRing::Inner(vec![
                Point::new(-0.51,-0.501),
                Point::new(-0.52, 0.502),
                Point::new( 0.53, 0.503),
                Point::new( 0.54,-0.504),
                Point::new(-0.51,-0.501),
        ])]);
        
        let converted_multipolygon = geo_types::MultiPolygon::<f64>::from(one_ring_polygon);

        let converted_polygon = converted_multipolygon.into_iter().next().unwrap();

        let expected_geotypes_polygon = geo_types::Polygon::new(
            LineString::from(vec![
                (-1.1,-1.01),
                (-1.2, 1.02),
                ( 1.3, 1.03),
                ( 1.4,-1.04),
                (-1.1,-1.01)
            ]), 
            vec![LineString::from(vec![
                (-0.51,-0.501),
                ( 0.54,-0.504), // NOTE: Direction gets reversed in conversion. Is this expected?
                ( 0.53, 0.503),
                (-0.52, 0.502),
                (-0.51,-0.501),
            ])],
        );

        assert_eq!(converted_polygon, expected_geotypes_polygon);
    }
```

### CODE REVIEW

**Comment by tmontaigu** [comment]
The inversion of the point order of inner rings is expected and is explained here https://docs.rs/shapefile/0.2.0/shapefile/record/polygon/enum.PolygonRing.html

**Comment by rfilmyer** [comment]
Ah, yep. I just checked the spec and it's there. I'll add some tests to explicitly check this behavior.

>A polygon consists of one or more rings. A ring is a connected sequence of four or more points that form a closed, nonself-intersecting loop. A polygon may contain multiple outer rings. *The order of vertices or orientation for a ring indicates which side of the ring is the interior of the polygon.* The neighborhood to the right of an observer walking along the ring in vertex order is the neighborhood inside the polygon. Vertices of rings defining holes in polygons are in a counterclockwise direction. Vertices for a single, ringed polygon are, therefore, always in clockwise order. The rings of a polygon are referred to as its parts.

[Page 8 of the original spec from 1998 (PDF)](https://www.esri.com/library/whitepapers/pdfs/shapefile.pdf)

**Comment by rfilmyer** [comment]
Alright, anything else I should do to get this PR into a mergeable state?

File: Cargo.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ exclude = ["tests/data/*"]
 [dependencies]
 byteorder = "1.2.7"
 dbase = "0.0.4"
-geo-types = {version = "0.4.3", optional = true}
+geo-types = {version = ">=0.4.0, <=0.6.0", optional = true}
```
Comment (tmontaigu) [inline Cargo.toml:17]: I think it should be ">=0.4.0, <0.7.0" so that bug fixes releases of geo-types 0.6.x qualify as compatible

**Pull request merged**